test/index.test.js

```diff
@@ -16,300 +18,442 @@ setGlobalDispatcher(new Agent({
   }
 }))
 
-test('restart fastify', async ({ pass, teardown, plan, same, equal }) => {
-  plan(11)
-
-  const _opts = {
-    port: 0,
-    app: myApp
-  }
+test('should create and restart fastify app', async (t) => {
+  async function createApplication (fastify, opts) {
+    const app = fastify(opts)
 
-  async function myApp (app, opts) {
-    pass('application loaded')
-    equal(opts, _opts)
     app.get('/', async () => {
       return { hello: 'world' }
     })
-  }
 
-  const server = await start(_opts)
+    return app
+  }
 
-  same(server.app.restarted, false)
+  const app = await restartable(createApplication)
 
-  const { stop, restart, listen } = server
-  teardown(stop)
+  t.teardown(async () => {
+    await app.close()
+  })
 
-  const { port, address } = await listen()
+  const host = await app.listen({ host: '127.0.0.1', port: 5843 })
+  t.equal(host, 'http://127.0.0.1:5843')
+  t.equal(app.addresses()[0].address, '127.0.0.1')
+  t.equal(app.addresses()[0].port, 5843)
 
-  equal(address, '127.0.0.1')
-  equal(port, server.port)
-  equal(address, server.address)
+  t.equal(app.restarted, false)
 
   {
-    const res = await request(`http://127.0.0.1:${port}`)
-    same(await res.body.json(), { hello: 'world' })
+    const res = await request(host)
+    t.same(await res.body.json(), { hello: 'world' })
   }
 
-  await restart()
-  same(server.app.restarted, true)
+  await app.restart()
+  t.same(app.restarted, true)
 
   {
-    const res = await request(`http://127.0.0.1:${port}`)
-    same(await res.body.json(), { hello: 'world' })
+    const res = await request(host)
+    t.same(await res.body.json(), { hello: 'world' })
   }
 })
 
-test('https', async ({ pass, teardown, plan, same, equal }) => {
-  plan(5)
+test('should create and restart fastify app twice', async (t) => {
+  async function createApplication (fastify, opts) {
+    const app = fastify(opts)
 
-  async function myApp (app, opts) {
-    pass('application loaded')
-    app.get('/', async (req, reply) => {
+    app.get('/', async () => {
       return { hello: 'world' }
     })
+
+    return app
   }
 
-  const { listen, stop, restart } = await start({
-    port: 0,
-    protocol: 'https',
-    key: await readFile(path.join(__dirname, 'fixtures', 'key.pem')),
-    cert: await readFile(path.join(__dirname, 'fixtures', 'cert.pem')),
-    app: myApp
+  const app = await restartable(createApplication)
+
+  t.teardown(async () => {
+    await app.close()
   })
-  teardown(stop)
 
-  const { address, port } = await listen()
+  const host = await app.listen({ host: '127.0.0.1', port: 5844 })
```

Comment:
I can extract them into the variable if that what you mean, but I can't set it to 0 like in other tests because here I'm testing that restartable fastify starts server on the port I pass.

Reply:
done